components/tikv_util/src/config.rs

```diff
@@ -425,15 +425,23 @@ pub fn canonicalize_path(path: &str) -> Result<String, Box<dyn Error>> {
 }
 
 pub fn canonicalize_sub_path(path: &str, sub_path: &str) -> Result<String, Box<dyn Error>> {
-    let parent = Path::new(path);
-    let p = parent.join(Path::new(sub_path));
+    let mut p = Path::new(path).to_path_buf();
+    if !sub_path.is_empty() {
```

Comment:
Why need to check if it's empty?

Reply:
If `sub_path ` is empty and `let p2 = p1.join(sub_path)`, `p2` will be a dir path which is different with `p1` and `p2.is_file()` will return `false`, even `p1` is exist and is a file!